src/set.rs

```diff
@@ -0,0 +1,913 @@
+use crate::raw::{self, InsertResult};
+use seize::{Collector, Guard, LocalGuard, OwnedGuard};
+
+use crate::map::ResizeMode;
+use std::borrow::Borrow;
+use std::collections::hash_map::RandomState;
+use std::fmt;
+use std::hash::{BuildHasher, Hash};
+use std::marker::PhantomData;
+
+/// A concurrent hash set.
+///
+/// Most hash set operations require a [`Guard`](crate::Guard), which can be acquired through
+/// [`HashSet::guard`] or using the [`HashSet::pin`] API. See the [crate-level documentation](crate#usage)
+/// for details.
+pub struct HashSet<K, S = RandomState> {
+    raw: raw::HashMap<K, (), S>,
+}
+
+// Safety: We only ever hand out &K/V through shared references to the map,
+// so normal Send/Sync rules apply. We never expose owned or mutable references
+// to keys or values.
+// TODO
```

Comment:
Can this `TODO` be removed? The bounds look correct to me.

Reply:
Removed! Sorry for the late reply